src/components/NewTodo/NewTodo.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {
+  useRef,
+  useState,
+  Dispatch,
+  SetStateAction,
+  useCallback,
+} from 'react';
+import { createTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
+type Props = {
+  user: User | null
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+  setErrorNotification: (value: string) => void;
+  setIsShownTempTodo: (value: boolean) => void;
+  setPreviewTitle: (value: string) => void;
+
+};
+
+export const NewTodo: React.FC<Props> = ({
+  user,
+  setTodos,
+  setErrorNotification,
+  setIsShownTempTodo,
+  setPreviewTitle,
+}) => {
+  const [title, setTitle] = useState('');
+  const newTodoField = useRef<HTMLInputElement>(null);
+
+  const handledTitle = useCallback(
+    (event:React.ChangeEvent<HTMLInputElement>) => {
+      const newTitle = event.target.value.replace(/^(\s)*/g, '');
+
+      setTitle(newTitle);
+      setPreviewTitle(newTitle);
+    }, [],
+  );
+
+  const createTodos = async () => {
+    setIsShownTempTodo(true);
+    try {
+      if (user) {
+        const newTodo = await createTodo(title, user?.id);
+
+        setTodos(prevTodos => [...prevTodos, newTodo]);
+      }
+    } catch (error) {
+      setErrorNotification('Unable to add a todo');
```

Comment:
you could create enum for error messages , this is a good practice to not rely on unknown string